tests/components/demo/test_config_flow.py

```diff
@@ -0,0 +1,100 @@
+"""Test the Demo config flow."""
+
+from collections.abc import AsyncGenerator
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant import config_entries, setup
+from homeassistant.components.demo import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture
+async def no_platforms() -> AsyncGenerator[None, None]:
```

Comment:
What is the advantage of `async def` here?
```suggestion
def no_platforms() -> Generator[None, None, None]:
```